shopinvader_image_webp/models/shopinvader_image_mixin.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+
+import base64
+from io import BytesIO
+
+from odoo import api, fields, models
+from odoo.addons.base_sparse_field.models.fields import Serialized
+from PIL import Image
+
+
+class ShopinvaderImageMixin(models.AbstractModel):
+    _inherit = "shopinvader.image.mixin"
+
+    images_webp = Serialized(
+        compute="_compute_images_webp", string="Shopinvader Image WebP"
+    )
+    images_webp_stored = Serialized()
+    images_webp_store_hash = fields.Char()
+
+    @api.depends("images")
+    def _compute_images_webp(self):
+        # Force computation if needed
+        self.filtered(
+            lambda x: x._images_webp_must_recompute()
+        )._compute_images_webp_stored()
+        for record in self:
+            record.images_webp = record.images_webp_stored
+
+    def _compute_images_webp_stored(self):
+        super()._compute_images_stored()
+        for record in self:
+            record.images_webp_stored = (
+                record._get_image_webp_data_for_record()
+            )
+            record.images_webp_store_hash = record._get_images_store_hash()
+
+    def _images_webp_must_recompute(self):
+        return self.images_webp_store_hash != self._get_images_store_hash()
+
+    def _create_storage_image(self, image, url_key):
```

Comment:
You can add `@api.model` for this function.
```suggestion
    @api.model
    def _create_storage_image(self, image, url_key):
```